sigopt/xgboost/experiment.py

```diff
@@ -66,19 +77,51 @@ def parse_and_create_metrics(self):
         else:
           metric['name'] = DEFAULT_EVALS_NAME + '-' + metric['name']
 
+  def check_and_fill_parameter_types(self):
+    params_to_check = [p for p in self.experiment_config_parsed['parameters'] if p['name'] in PARAMETER_INFORMATION]
+    for parameter in params_to_check:
+      parameter_name = parameter['name']
+      if parameter_name in PARAMETER_INFORMATION:  # TODO: if parameter_name isn't here, probably should throw warning
+        proper_parameter_type = PARAMETER_INFORMATION[parameter_name]['type']
+        if 'type' in parameter:
+          experiment_config_parameter_type = parameter['type']
+          if experiment_config_parameter_type != proper_parameter_type:
+            raise ValueError(f'Parameter {parameter_name} type listed incorrectly as {experiment_config_parameter_type}'
+                             f'in experiment config, and should be listed as having type {proper_parameter_type}.')
+        else:
+          parameter['type'] = proper_parameter_type
+
+  def check_and_fill_parameter_bounds(self):
+    params_to_check = [p for p in self.experiment_config_parsed['parameters'] if p['name'] in PARAMETER_INFORMATION]
+    for parameter in params_to_check:
+      parameter_name = parameter['name']
+      if 'bounds' not in parameter and PARAMETER_INFORMATION[parameter_name]['type'] in ['double', 'int']:
+        if parameter_name not in SUPPORTED_AUTOBOUND_PARAMS:
+          raise ValueError('We do not support autoselection of bounds for {param_name}.')
+        search_bound = SEARCH_BOUNDS[SUPPORTED_AUTOBOUND_PARAMS.index(parameter_name)]
```

Comment:
oh this is kinda brittle. Maybe we should rewrite `SEARCH_BOUNDS` to a dictionary

```
SEARCH_BOUNDS = {
  'param_name_1': {'name': 'param_name1', 'type': 'double', 'bounds': {'min': 0, 'max': 1}},
  'param_name_2': {'name': 'param_name2', 'type': 'double', 'bounds': {'min': 0, 'max': 1}},
  'param_name_3': {'name': 'param_name3', 'type': 'int', 'bounds': {'min': 0, 'max': 100}},
}
```
Now, instead of having two data structures referring to the "same thing" (aka the parameters we can auto-bound"), we just have one.

And line 101-102 can just simply be:
```
parameter.update(SEARCH_BOUNDS[parameter_name])
```